### PR TITLE
add imageRegistry method for addon to consume

### PR DESCRIPTION
**Pull request by zhiweiyin318**
Signed-off-by: Zhiwei Yin <zyin@redhat.com>

### CODE REVIEW

File: pkg/helpers/imageregistry/imageregistry.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+func NewImageRegistry(client client.Client) *ImageRegistryClient {
+	return &ImageRegistryClient{
+		client: client,
+	}
+}
+
+// Get returns the registry address and pullSecret.
+// found is true if the imageRegistry in the label open-cluster-management.io/image-registry of cluster exists.
+func (c *ImageRegistryClient) Get(clusterName string) (registryAddress, pullSecretNamespace, pullSecretName string, found bool, err error) {
```
Comment (qiujian16) [inline pkg/helpers/imageregistry/imageregistry.go:37]: why not merge Get and OverrideImage together?

Comment (zhiweiyin318) [reply pkg/helpers/imageregistry/imageregistry.go]: refactor to 4 methods:
GetPullSecret(clusterName string) (pullSecretNamespace, pullSecretName string, err error)
GetRegistry(clusterName string) (registry string, err error) 
OverrideImage(clusterName, imageName string) (newImageName string, err error) 
GetImageRegistry(clusterName string) (*v1alpha1.ManagedClusterImageRegistry, error)

File: pkg/helpers/imageregistry/imageregistry.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+		return err
+	}
+	return nil
+}
+
+type ImageRegistryClient struct {
+	client client.Client
+}
+
+func NewImageRegistry(client client.Client) *ImageRegistryClient {
```
Comment (qiujian16) [inline pkg/helpers/imageregistry/imageregistry.go:30]: it will not be easy if the consumer is not using  controller-runtime

Comment (zhiweiyin318) [reply pkg/helpers/imageregistry/imageregistry.go]: referring your suggestions, I define a Client interface and create 2 objects, one uses controller-time, another uses kubeconfig.

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**